src/modules/test/MimParser.cpp

```diff
@@ -97,4 +86,91 @@ pMimObjects MimParser::ParseMim(std::string path)
 
     json_value_free(root_value);
     return mimObjects;
+}
+
+void MimParser::ParseMimField(JSON_Object* jsonField, MimObject& mimObject)
+{
+    // TODO: this should be the root for parsing all MimObjects?
+    MimField mimField;
+    std::cout << "Parsing field: " << json_object_get_string(jsonField, "name") << std::endl;
+    if (json_object_has_value_of_type(jsonField, "schema", JSONString) ||
+        json_object_has_value_of_type(jsonField, "schema", JSONNumber) ||
+        json_object_has_value_of_type(jsonField, "schema", JSONBoolean))
+    {
+        mimField = {
+            json_object_get_string(jsonField, "name"),
+            json_object_get_string(jsonField, "schema"),
+            "", // subType1
+            "", // subType2
+            std::make_shared<std::vector<std::string>>()};
+    }
+    else if (json_object_has_value_of_type(jsonField, "schema", JSONObject))
+    {
+        // TODO: Turn into recursive function to support embedded objects?
+
+        JSON_Object *jsonSchema = json_object_get_object(jsonField, "schema");
+
+        if (0 == strcmp(json_object_get_string(jsonSchema, "type"), "enum"))
+        {
+            mimField = {
+                json_object_get_string(jsonField, "name"),
+                json_object_get_string(jsonSchema, "valueSchema"),
+                "", // subType1
+                "", // subType2
+                std::make_shared<std::vector<std::string>>()};
+
+            // Add supported values
+            if (json_object_has_value_of_type(jsonSchema, "enumValues", JSONArray))
+            {
+                JSON_Array *supportedValues = json_object_get_array(jsonSchema, "enumValues");
+                for (size_t a = 0; a < json_array_get_count(supportedValues); a++)
+                {
+                    JSON_Object *jsonEnumValues = json_array_get_object(supportedValues, a);
+                    mimField.allowedValues->push_back(std::to_string(json_object_get_number(jsonEnumValues, "enumValue")));
+                }
+            }
+        }
+        else if (0 == strcmp(json_object_get_string(jsonSchema, "type"), "array"))
+        {
+            std::cout << "Name: " << json_object_get_string(jsonField, "name") << ", elementSchema: " << json_object_dotget_string(jsonField, "schema.elementSchema") << std::endl;
+            if (json_object_has_value_of_type(jsonSchema, "elementSchema", JSONObject))
+            {
+                // Array of objects
```

Comment:
>// Array of objects [](http://example.com/codeflow?start=16&length=19)

There is no array of objects in the MIM except the root of the MIM. Is that what this is?

Reply:
Its an array of MimSettings, ill change the datatype and remove the comment, its descriptive enough